test/sample-test/run_test.sh

```diff
@@ -101,242 +103,236 @@ while [ "$1" != "" ]; do
                                                 ;;
              * )                                usage
                                                 exit 1
-    esac
-    shift
+  esac
+  shift
 done
 
-if [ -z "$RESULTS_GCS_DIR" ]; then
-    usage
-    exit 1
+GITHUB_REPO=kubeflow/pipelines
+BASE_DIR=/python/src/github.com/${GITHUB_REPO}
+TEST_DIR=${BASE_DIR}/test/sample-test
+
+#TODO(numerology): For utility functions, check the paths after gaoning777
+# merges his PR.
+################################################################################
+# Utility function to setup working dir, input and output locations.
+# Globals:
+#   SAMPLE_TEST_RESULT, test output file.
+#   SAMPLE_TEST_OUTPUT, test output location.
+# Arguments:
+#   Test name.
+################################################################################
+preparation() {
+  SAMPLE_TEST_RESULT="junit_Sample$1Output.xml"
+  SAMPLE_TEST_OUTPUT=${RESULTS_GCS_DIR}
+  WORK_DIR=${BASE_DIR}/samples/core/$1
+  cd ${WORK_DIR}
+}
+
+################################################################################
+# Utility function to generate formatted result, and move it to a place for PROW
+# to pick up.
+# Arguments:
+#   Test name.
+################################################################################
+check_result() {
+  cd "${TEST_DIR}"
+  #TODO: basic tests use .zip but other use .yaml, need consistency. Plan to
+  # unify them into .yaml.
+  python3 run_sample_test.py --input ${WORK_DIR}/$1.yaml \
+  --result ${SAMPLE_TEST_RESULT} --output ${SAMPLE_TEST_OUTPUT} --testname $1 \
+  --namespace ${NAMESPACE}
+
+  echo "Copy the test results to GCS ${RESULTS_GCS_DIR}/"
+  gsutil cp ${SAMPLE_TEST_RESULT} ${RESULTS_GCS_DIR}/${SAMPLE_TEST_OUTPUT}
+}
+
+################################################################################
+# Utility function to prepare and validate the results for sample test based on
+# ipynb.
+# Arguments:
+#   Test name.
+################################################################################
+check_notebook_result() {
+  jupyter nbconvert --to python $1.ipynb
+  pip3 install tensorflow==1.8.0
```

Comment:
We might conserve time is we integrate tensorflow in our test image as opposed to dynamically installing it.